beacon-chain/rpc/beacon_server.go

```diff
@@ -244,6 +243,16 @@ func (bs *BeaconServer) PendingDeposits(ctx context.Context, _ *ptypes.Empty) (*
 		return nil, fmt.Errorf("could not generate historical deposit trie from deposits: %v", err)
 	}
 
+	allPendingDeps := bs.beaconDB.PendingDeposits(ctx, bNum)
```

Comment:
Is this sorted? If it's not sorted then your logic below will not work. 

I would not count on this being sorted.

Reply:
It's sorted in the PendingDeposits function by merkle index.